api/process_visual_generation.py

```diff
@@ -0,0 +1,245 @@
+"""
+# 📊 Process Visualization Module
+
+Module for visualizing process event logs using **PM4Py**.
+
+---
+
+## Features
+
+This module provides functions to:
+
+- Extract **start** and **end** activities from a CSV log.
+- Generate and save various visualizations:
+  - **Process tree**
+  - **Process graph (with filtering)**
+  - **Performance-based DFG** (Directed Flow Graph)
+  - **Occurrence-based DFG** (Directed Flow Graph)
+  - **Petri net**
+
+---
+
+## Dependencies
+
+- `pandas`
+- `pm4py`
+
+---
+
+## Example
+
+```python
+from process_visual_generation import start_end_activities, generate_tree_inductive
+
+start_acts, end_acts = start_end_activities("data/log.csv")
+generate_tree_inductive("data/log.csv", "~/Downloads")
+```
+"""
+
+import os
+from api.io_helpers import generate_filename, load_event_log
+import pandas
+import pm4py
+
+
+def start_end_activities(csv_path):
+    """
+    Reads an event log from a CSV file and returns its start and end activities.
+
+    Assumes the CSV has the following columns:
```

Comment:
Reference the python function that creates the .csv instead, and that function contains the specification of the csv returned. Have the user know what function to use, rather than explain it as if they will code it! Remember Kaiaulu has \code{\link{function_name}} in R. Find the equivalent in python. 

You should also see itm0 for the "See Also" section so you can reference on the returns what functions will use the output of another.